gpflow/utilities/utilities.py

```diff
@@ -246,22 +247,123 @@ def clear_bijector(bijector, _, state):
     return input_module
 
 
-M = TypeVar('M', bound=tf.Module)
+def _get_by_name_index(parent: object, attr_str: str, index_str: str) -> object:
+    attr = getattr(parent, attr_str)
+    if index_str != "":
+        index = int(index_str)
+        return attr[index]
+    return attr
+
+
+def _set_by_name_index(parent: object, value: Any, attr_str: str, index_str: str):
+    if index_str != "":
+        index = int(index_str)
+        attr = getattr(parent, attr_str)
+        attr[index] = value
+    else:
+        setattr(parent, attr_str, value)
+
+
+def _get_last_attr_spec(parent: object, attr_path: str) -> Tuple[object, str, str]:
```

Comment:
you mention that it doesn't handle paths like `attr[0].attr.attr[-1]`

  - what are the implications of that?
  - if we don't handle that, can we be explicit about it in the docs?
  - is the docstring example if we don't handle this case?

Reply:
more generally, it may be much safer to generate a lambda function that sets the value, then pass that the value

Reply:
It does handle this case. Where did you get it from?

Reply:
err ... your comment at the top of the PR

Reply:
> you mention that it doesn't handle paths like `attr[0].attr.attr[-1]`

I believe that is referring to the original PR (#1266)